app/src/androidTest/java/org/kiwix/kiwixmobile/main/TopLevelDestinationTest.kt

```diff
@@ -33,40 +35,51 @@ class TopLevelDestinationTest : BaseActivityTest() {
     PreferenceManager.getDefaultSharedPreferences(context).edit {
       putBoolean(SharedPreferenceUtil.PREF_SHOW_INTRO, false)
       putBoolean(SharedPreferenceUtil.PREF_WIFI_ONLY, false)
+      putBoolean(SharedPreferenceUtil.PREF_IS_TEST, true)
+      putBoolean(SharedPreferenceUtil.PREF_EXTERNAL_LINK_POPUP, true)
     }
   }
 
   @Test
   fun testTopLevelDestination() {
-    topLevel {
-      clickReaderOnBottomNav {
-      }
-      clickLibraryOnBottomNav {
-        assertGetZimNearbyDeviceDisplayed()
-        clickFileTransferIcon {
+    if (Build.VERSION.SDK_INT < Build.VERSION_CODES.TIRAMISU) {
+      topLevel {
+        clickReaderOnBottomNav {
         }
-      }
-      clickDownloadOnBottomNav {
-        assertLibraryListDisplayed()
-        clickOnGlobeIcon {
+        clickLibraryOnBottomNav {
+          assertGetZimNearbyDeviceDisplayed()
+          clickFileTransferIcon {
+          }
         }
+        clickDownloadOnBottomNav {
+          assertLibraryListDisplayed()
+          clickOnGlobeIcon {
+          }
+        }
+        clickBookmarksOnNavDrawer {
+          assertBookMarksDisplayed()
+          clickOnTrashIcon()
+          assertDeleteBookmarksDialogDisplayed()
+        }
+        clickHistoryOnSideNav {
+          assertHistoryDisplayed()
+          clickOnTrashIcon()
+          assertDeleteHistoryDialogDisplayed()
+        }
+        clickHostBooksOnSideNav(ZimHostRobot::assertMenuWifiHotspotDiplayed)
+        clickSettingsOnSideNav(SettingsRobot::assertMenuSettingsDisplayed)
+        clickHelpOnSideNav(HelpRobot::assertToolbarDisplayed)
+        clickSupportKiwixOnSideNav()
+        assertExternalLinkDialogDisplayed()
+        pressBack()
       }
-      clickBookmarksOnNavDrawer {
-        assertBookMarksDisplayed()
-        clickOnTrashIcon()
-        assertDeleteBookmarksDialogDisplayed()
-      }
-      clickHistoryOnSideNav {
-        assertHistoryDisplayed()
-        clickOnTrashIcon()
-        assertDeleteHistoryDialogDisplayed()
-      }
-      clickHostBooksOnSideNav(ZimHostRobot::assertMenuWifiHotspotDiplayed)
-      clickSettingsOnSideNav(SettingsRobot::assertMenuSettingsDisplayed)
-      clickHelpOnSideNav(HelpRobot::assertToolbarDisplayed)
-      clickSupportKiwixOnSideNav()
-      assertExternalLinkDialogDisplayed()
-      pressBack()
+    }
+  }
+
+  @After
+  fun setIsTestPreference() {
+    PreferenceManager.getDefaultSharedPreferences(context).edit {
+      putBoolean(SharedPreferenceUtil.PREF_IS_TEST, false)
```

Comment:
Does this actually change the main preference of users? If yes, we shouldn't do this.

Reply:
Its only used in test cases  , why i have placed it because of by passing ``dialog popup for requesting MANAGE_EXTERNAL_PERMISSION``` , which is not need during testing and I have observed many times this dialog remain on screen leads to test failure on android 30.